tests/integration/tier0/non-destructive/application-lock/test_application_lock.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+
+
+@pytest.mark.test_simultaneous_runs
+def test_simultaenous_runs(convert2rhel):
+    """
+    Verify that running convert2rhel locks out other instances.
+    1/ Invoke convert2rhel, wait on data collection acknowledgement prompt.
+    2/ Invoke second instance of convert2rhel, observe warning and the utility exit.
+    3/ Exit the first run of convert2rhel.
+    4/ Invoke third instance of convert2rhel; with the previous instances dead, the third instance should be allowed to run.
+    5/ Exit the utility on the first prompt.
+    """
+    # Invoke a first instance
+    with convert2rhel("--no-rpm-va --debug") as c2r_one:
+        c2r_one.expect("Continue with the system conversion?")
+        # Invoke a second run
+        with convert2rhel("--no-rpm-va --debug") as c2r_two:
+            c2r_two.expect("Another copy of convert2rhel is running.")
+            assert c2r_two.exitstatus == 1
```

Comment:
For some reason, pexpect is not happy with us calling convert2rhel twice here. Maybe if instead of using convert2rhel fixture, you can call subprocess instead? 

I would think of an alternative here as: 
1. Get the PID of the first running process
2. Check that the file exists in the /var/run/... and check if that matches de PID
3. Start a subprocess call calling convert2rhel --help, for example
4. Assert the message was sent to stderr and the exit code was 1
5. Stop the process sending the letter `N` as you did there
6. Call convert2rhel fixture as a new process, making it the third run. 

What do you think?

Reply:
Me and @jochapma talked over slack and he will try implementing my suggestion first to see if that will work.